core/domain/user_jobs_continuous.py

```diff
@@ -293,100 +291,72 @@ def _handle_incoming_event(cls, active_realtime_layer, event_type, *args):
-        rating_term = average_rating - cls.MIN_AVERAGE_RATING
-        # Only explorations with an average rating greater than the minimum
-        # have an impact.
-        if rating_term <= 0:
-            return 0
-
-        # Get num_ratings_scaler.
-        if total_num_ratings < cls.NUM_RATINGS_SCALER_CUTOFF:
-            num_ratings_scaler = cls.NUM_RATINGS_SCALER * total_num_ratings
-        else:
-            num_ratings_scaler = 1.0
-
-        # Get number of completions/playthroughs.
-        num_completions = (
-            stats_jobs_continuous.StatisticsAggregator.get_statistics(
-                exploration_id, stats_jobs_continuous.VERSION_ALL)[
-                    'complete_exploration_count'])
-        # Only take the log of a non-zero number.
-        if num_completions <= 0:
-            return 0
-        num_completions_term = math.log(num_completions)
-
-        exploration_impact_score = (
-            rating_term *
-            num_completions_term *
-            num_ratings_scaler *
-            cls.MULTIPLIER
-        )
-
-        return exploration_impact_score
+        return [exp_models.ExpSummaryModel]
 
     @staticmethod
     def map(item):
         if item.deleted:
             return
 
-        exploration_impact_score = (
-            UserImpactMRJobManager._get_exp_impact_score(item.id))
+        exponent = 2/float(3)
 
-        if exploration_impact_score > 0:
-            # Get exploration summary and contributor ids,
-            # yield for each contributor.
+        # Get average rating and value per user
+        total_rating = 0
+        for ratings_value in item.ratings:
+            total_rating += item.ratings[ratings_value] * int(ratings_value)
+        if sum(item.ratings.itervalues()):
+            average_rating = total_rating / sum(item.ratings.itervalues())
+            value_per_user = average_rating - 2
+        else:
+            value_per_user = 0
+
+        statistics = (
+            stats_jobs_continuous.StatisticsAggregator.get_statistics(item.id,
+                                                                      'all'))
+        answer_count = 0
+        # Find number of users per state (card), and subtract no answer
+        # This will not count people who have been back to a state twice
+        # but did not give an answer the second time, but is probably the
+        # closest we can get with current statistics to "number of users
+        # who gave an answer" since it is "number of users who always gave
+        # an answer".
+        for state_name in statistics['state_hit_counts']:
+            state = statistics['state_hit_counts'][state_name]
+            first_entry_count = state['first_entry_count']
+            no_answer_count = state['no_answer_count']
+            answer_count += first_entry_count - no_answer_count
+        # Turn answer count into reach
+        reach = answer_count**exponent
+
+        if value_per_user > 0:
```

Comment:
You actually would need it because of the -2 from the average rating. But I also added a short cut earlier and was able to remove this anyway, that way it was able to short cut as much calculation as possible.